internal/util/streamingutil/service/discoverer/channel_assignment_discoverer.go

```diff
@@ -58,7 +58,10 @@ func (d *channelAssignmentDiscoverer) parseState() VersionedState {
 	for _, assignment := range d.lastDiscovery.Assignments {
 		assignment := assignment
 		addrs = append(addrs, resolver.Address{
-			Addr:               assignment.NodeInfo.Address,
+			Addr: assignment.NodeInfo.Address,
+			// resolverAttributes is important to use when resolving, server id to make resolver.Address with same adresss different.
+			Attributes: attributes.WithServerID(new(attributes.Attributes), assignment.NodeInfo.ServerID),
```

Comment:
Attach the serverID as a resolver attributes to avoid lost address discovery when reusing address of milvus node.